File: VB/DiagramDataBindingBehavior/Data/ClassData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DiagramDataBindingBehavior.Data
	Public Class ClassData
		Public Property ClassName() As String
		Public Property Type() As ClassType
	End Class
	Public Enum ClassType
		[Class]
		[Interface]
		AbstractClass
	End Enum

	Public Class ConnectionData
		Public Property ConnectedFrom() As Object
		Public Property ConnectedTo() As Object
	End Class
End Namespace

File: CS/DiagramDataBindingBehavior/Data/ClassData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DiagramDataBindingBehavior.Data
{
    public class ClassData
    {
        public string ClassName { get; set; }
        public ClassType Type { get; set; }
    }
    public enum ClassType
    {
        Class,
        Interface,
        AbstractClass,
    }

    public class ConnectionData
    {
        public object ConnectedFrom { get; set; }
        public object ConnectedTo { get; set; }
    }
}
